dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataSqlQueryFactoryBuilder.java

```diff
@@ -0,0 +1,288 @@
+  // For Terdata version <= 14
+  private boolean isOldVersion;
+  private final boolean isLSql;
+
+  /* pp */ static TeradataSqlQueryFactoryBuilder startBuildingFrom(
+      ConnectorArguments arguments, boolean forLSql) throws MetadataDumperUsageException {
+    TeradataSqlQueryFactoryBuilder builder = new TeradataSqlQueryFactoryBuilder(forLSql);
+    builder.parse(arguments);
+    return builder;
+  }
+
+  /* pp */ static TeradataSqlQueryFactoryBuilder startBuildingFrom(ConnectorArguments arguments)
+      throws MetadataDumperUsageException {
+    return startBuildingFrom(arguments, false);
+  }
+
+  /* pp */ TeradataSqlQueryFactoryBuilder within(ZonedInterval interval) {
+    this.interval = interval;
+    return this;
+  }
+
+  private TeradataSqlQueryFactoryBuilder(boolean isLSql) {
+    this.isLSql = isLSql;
+  }
+
+  private void parse(ConnectorArguments arguments) throws MetadataDumperUsageException {
+    isAssessment = arguments.isAssessment();
+    isOldVersion = arguments.getConnectorName().startsWith("teradata14");
+    parseAlternates(arguments.getQueryLogAlternates());
+    addConditions(arguments.getQueryLogEarliestTimestamp());
+    addOrderColumns();
+    selectExpressions();
+  }
+
+  private void parseAlternates(List<String> alternates) throws MetadataDumperUsageException {
+    logTable = isAssessment ? ASSESSMENT_DEF_LOG_TABLE : DEF_LOG_TABLE;
+    queryTable = DEF_QUERY_TABLE;
+    if (alternates != null && !alternates.isEmpty()) {
+      if (alternates.size() != 2)
+        throw new MetadataDumperUsageException(
+            "Alternate query log tables must be given as a pair; you specified: " + alternates);
+      logTable = alternates.get(0);
+      queryTable = alternates.get(1);
+    }
+  }
+
+  private void addConditions(String earliestTimestamp) {
+    conditions = new ArrayList<>();
+    // if the user specifies an earliest start time there will be extraneous empty dump files
+    // because we always iterate over the full 7 trailing days; maybe it's worth
+    // preventing that in the future. To do that, we should require getQueryLogEarliestTimestamp()
+    // to parse and return an ISO instant, not a database-server-specific format.
+    if (!StringUtils.isBlank(earliestTimestamp)) {
+      if (isLSql) {
+        conditions.add("ST.CollectTimeStamp >= " + earliestTimestamp);
+      } else {
+        conditions.add("L.StartTime >= " + earliestTimestamp);
+      }
+    }
+    if (!(isOldVersion || isAssessment)) conditions.add("L.UserName <> 'DBC'");
+  }
+
+  private void addOrderColumns() {
+    orderColumns =
+        isOldVersion || !isAssessment
+            ? Collections.emptyList()
+            : Arrays.asList("ST.QueryID", "ST.SQLRowNo");
+  }
+
+  private void selectExpressions() {
+    if (isLSql) expressions = EXPRESSIONS_LSQL_TBL;
+    else if (isOldVersion) expressions = EXPRESSIONS_LOG_TBL;
+    else {
+      String[] selections = isAssessment ? EXPRESSIONS_FOR_ASSESSMENT : EXPRESSIONS;
```

Comment:
I would suggest to change the type of `EXPRESSIONS_FOR_ASSESSMENT` and `EXPRESSIONS` to `ImmutableList<String>`.